src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joined = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== 0) {
+        joined += separator;
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        continue;
+      }
+
+      joined += this[i];
```

Comment:
`continue` instruction can be really confusing, I suggest you to just invert your condition to get rid of it